Enhance pre-commit hook to automate migrations ref doc and revision metadata (#21601)
mistercrunch/Airflow: docs/apache-airflow/migrations-ref.rst

Here's the list of all the Database Migrations that are executed via when you run ``airflow db upgrade``:

 .. This table is automatically updated by pre-commit by ``scripts/ci/pre_commit/pre_commit_migration_documented.py``
 .. Changes should be made to config file ``airflow/migrations/migrations-info.yaml``
 .. Beginning of auto-generated table

> 

Author: one reason the diff is ugly is i wrap a little more aggressively

but also the column widths end up being a little different when using `tabulate` and wasn't having luck getting it to conform so left it with the ugly diff

Here's the list of all the Database Migrations that are executed via when you run ``airflow db upgrade``:

 .. This table is automatically updated by pre-commit by ``scripts/ci/pre_commit/pre_commit_revision_map.py``

>  .. Changes should be made to config file ``airflow/migrations/migrations-info.yaml``

Reviewer: ```suggestion
 .. Changes should be made to in the migration files themselves
```

|                                 |                   |             | table                                                        |
+---------------------------------+-------------------+-------------+--------------------------------------------------------------+
| ``13eb55f81627``                | ``1507a7289a2f``  | ``1.5.0``   | Maintain history for compatibility with earlier migrations   |
+---------------------------------+-------------------+-------------+--------------------------------------------------------------+
| ``1507a7289a2f``                | ``e3a246e0dc1``   | ``1.5.0``   | Add ``is_encrypted`` column in ``connection`` table          |
+---------------------------------+-------------------+-------------+--------------------------------------------------------------+
| ``e3a246e0dc1`` (base)          | ``None``          | ``1.5.0``   | current schema                                               |
+---------------------------------+-------------------+-------------+--------------------------------------------------------------+

 .. End of auto-generated table

> 
> >>>>>>> c061b7f4b (Use yaml config for generation of migration reference)

Reviewer: ```suggestion
```

Rebase leftovers?
